hudi-spark-datasource/hudi-spark/src/test/resources/upgrade-downgrade-fixtures/generate-fixtures.sh

```diff
@@ -0,0 +1,341 @@
+    else
+        # Use Maven packages for official releases
+        local hudi_bundle="org.apache.hudi:hudi-spark${spark_version}-bundle_${scala_version}:${hudi_version}"
+        echo "Using Hudi bundle: $hudi_bundle"
+        "$spark_home/bin/spark-shell" \
+            --conf 'spark.serializer=org.apache.spark.serializer.KryoSerializer' \
+            --conf 'spark.sql.catalog.spark_catalog=org.apache.spark.sql.hudi.catalog.HoodieCatalog' \
+            --conf 'spark.sql.extensions=org.apache.spark.sql.hudi.HoodieSparkSessionExtension' \
+            --conf "spark.jars.ivy=$ivy_cache_dir" \
+            --conf 'spark.sql.warehouse.dir=/tmp/spark-warehouse' \
+            --packages "$hudi_bundle" -i "$temp_script"
+    fi
+    
+    # Clean up ivy cache directory
+    rm -rf "$ivy_cache_dir"
+    
+    # Clean up temp script
+    rm "/tmp/generate_${fixture_name}.scala"
+    
+    echo "Fixture $fixture_name generated successfully"
+}
+
+# Generate fixtures for each version using appropriate Spark versions
+echo "Generating fixtures for all supported versions..."
+
+# Based on Hudi-Spark compatibility matrix:
+# 1.0.x    -> 3.5.x (default)
+# 0.15.x   -> 3.5.x (default)
+# 0.14.x   -> 3.4.x (default)
+# 0.13.x   -> 3.3.x (default)
+# 0.12.x   -> 3.3.x (default)
+# 0.11.x   -> 3.2.x (default)
+
+echo "Hudi Version -> Spark Version -> Scala Version mapping:"
+
+# Hudi 0.11.1 (Table Version 4) -> Spark 3.2.x (default) -> Scala 2.12
+if should_generate_version "4"; then
+    echo "   0.11.1 -> Spark 3.2 -> Scala 2.12"
+    generate_fixture "0.11.1" "4" "hudi-v4-table" "3.2" "2.12"
+fi
+
+# Hudi 0.12.2 (Table Version 5) -> Spark 3.3.x (default) -> Scala 2.12  
+if should_generate_version "5"; then
+    echo "   0.12.2 -> Spark 3.3 -> Scala 2.12"
+    generate_fixture "0.12.2" "5" "hudi-v5-table" "3.3" "2.12"
+fi
+
+# Hudi 0.14.0 (Table Version 6) -> Spark 3.4.x (default) -> Scala 2.12
+if should_generate_version "6"; then
+    echo "   0.14.0 -> Spark 3.4 -> Scala 2.12"
+    generate_fixture "0.14.0" "6" "hudi-v6-table" "3.4" "2.12"
+fi
+
+# Hudi 1.0.2 (Table Version 8) -> Spark 3.5.x (default) -> Scala 2.12
+if should_generate_version "8"; then
+    echo "   1.0.2 -> Spark 3.5 -> Scala 2.12"
+    generate_fixture "1.0.2" "8" "hudi-v8-table" "3.5" "2.12"
+fi
+
+# Hudi 1.1.0 (Table Version 9) -> Spark 3.5.x (default) -> Scala 2.12 (requires local bundle)
+if should_generate_version "9"; then
+    echo "   1.1.0 -> Spark 3.5 -> Scala 2.12 (using local bundle)"
+    generate_fixture "1.1.0" "9" "hudi-v9-table" "3.5" "2.12"
+fi
+
+
+echo ""
+echo "Fixture generation completed!"
+
+# Compress fixture tables to save space
+echo ""
+echo "Compressing fixture tables..."
+for fixture_dir in "$FIXTURES_DIR"/hudi-v*-table; do
+    if [ -d "$fixture_dir" ]; then
+        fixture_name=$(basename "$fixture_dir")
+        echo "Compressing $fixture_name..."
+        (cd "$FIXTURES_DIR" && zip -r -q -X "${fixture_name}.zip" "$fixture_name")
```

Comment:
See if the `zip` command can excluded unnecessary files as mentioned in the other review comment.

Reply:
Will do so